Allows resting people to get off the ground w/ assistance.

Cameron653 opened the pull request:
This allows it so resting mobs/humans can grab onto another person to hoist theirself up off the ground.

Belsima commented:
Maybe use a gender check for himself/herself/themself- theirself doesn't seem right at all. Also maybe a very slight delay for getting up?

Anewbe commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -258,7 +258,16 @@
+							"<span class='notice'>You grip onto [src] and pull yourself up off the ground!</span>")
+					M.resting = 0 //Hoist yourself up up off the ground. No para/stunned/weakened removal.
+					if(M.fire_stacks >= (src.fire_stacks + 3)) //Fire checks.
+						src.fire_stacks += 1
```
adjust_fire_stacks(1)

Anewbe commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -258,7 +258,16 @@
+					M.resting = 0 //Hoist yourself up up off the ground. No para/stunned/weakened removal.
+					if(M.fire_stacks >= (src.fire_stacks + 3)) //Fire checks.
+						src.fire_stacks += 1
+						M.fire_stacks -= 1
```
adjust_fire_stacks(-1)

Cameron653 commented:
Ah, alright. I went ahead and copied the code below it for the firestacks, so I'll update it for both. And this is intended to not have a delay, as it's supposed to be an alternative to typing in rest, which does have a delay.

Cameron653 commented:
There we go. Adjusted it to use adjust_fire_stacks and made it check their shown gender.

Anewbe commented:
I'm gonna ask that you give it a delay, if you don't mind.

Cameron653 commented:
Alright. Added a delay and fixed the gender check by using \himself since
https://github.com/PolarisSS13/Polaris/blob/master/code/modules/mob/living/carbon/human/human.dm?utf8=%E2%9C%93#L865
something like the above line didn't work.

Anewbe commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -259,14 +259,15 @@
 						M.adjust_fire_stacks(-1)
 					if(M.on_fire)
 						src.IgniteMob()
+					sleep(5) //.5 second delay. Makes it a bit stronger than just typing rest.
```
Genuinely unsure, but does a do_after not work here?

Cameron653 replied:
I'm not sure. I just used sleep as I've seen it used many times before and I know it'd work here. I can see if do_after works.

Anewbe replied:
It may not work, I don't know if do_after() works for resting mobs.

Cameron653 commented:
Changed the sleep to do_after.

Anewbe commented:
code/modules/mob/living/carbon/carbon.dm:269:error: user: undefined var

Cameron653 commented:
Fixed. Compiling to retest now.

Anewbe commented:
Also, please compile and test your code before you commit it. Not 100% certain, but that seems like something that the compiler would catch.

Cameron653 commented:
Yeah. I tested it initially before adding all the changes (do after, delay, \himself, etc) so was just waiting for feedback to be finished before giving it one last final test, since the compiler takes ages.

Anewbe commented:
If you swap the Northern Star Map for the Example map it goes much quicker. I learned this only yesterday.

Cameron653 commented:
Alright. Tested and can confirm everything works: https://i.imgur.com/lSwDwId.gifv

Anewbe commented:
Just had a thought: Can we add an intent check for the target, so people I push over can't just pick themselves up off of my unless I'm on help intent?

Cameron653 commented:
You can't pick yourself up if you're weakened, paralyzed, or have any other status debuff that prevents you from clicking. 

All this does is set resting to 0, so it wouldn't do anything if they were pushed over as they would be weakened and unable to click.

https://github.com/PolarisSS13/Polaris/blob/e158fcd3cc70e657bddba1a331f6b959441c417a/code/modules/mob/living/carbon/human/human_attackhand.dm#L291

A check for resting would be possible, but lines
https://github.com/Cameron653/Polaris/blob/b41c2fec9a1320c1d63dcd34c8670b587a11e85e/code/modules/mob/living/carbon/carbon.dm#L253-L257

should catch if the person being clicked on is resting, in which case they'd shake them awake.

Anewbe commented:
Alrighty, thank you for looking into that.
